modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/DocumentDataDefinitionTypeResourceTest.java

```diff
@@ -92,13 +99,49 @@ public void testPostSiteDocumentDataDefinitionTypes() throws Exception {
 
 		_assertDocumentDataDefinitionType(
 			null, ddmStructureIds,
-			documentDataDefinitionTypesResource.
-				postSiteDocumentDataDefinitionTypes(
+			documentDataDefinitionTypeResource.
+				postSiteDocumentDataDefinitionType(
 					testGroup.getGroupId(),
 					_createDocumentDataDefinitionType(ddmStructureIds)),
 			testGroup.getGroupId());
 	}
 
+	@Override
+	protected String[] getAdditionalAssertFieldNames() {
+		return new String[] {"availableLanguages", "description", "name"};
+	}
+
+	@Override
+	protected DocumentDataDefinitionType randomDocumentDataDefinitionType()
+		throws Exception {
+
+		DocumentDataDefinitionType documentDataDefinitionType =
+			super.randomDocumentDataDefinitionType();
+
+		documentDataDefinitionType.setAvailableLanguages(
+			() -> LocaleUtil.toW3cLanguageIds(
+				new Locale[] {LocaleUtil.getDefault()}));
+		documentDataDefinitionType.setDataLayout(new DataLayout());
+		documentDataDefinitionType.setDataDefinitionFields(
```

Comment:
Maybe setDataDefinitionFields should go before setDataLayout

Reply:
Done